src/nomad_simulations/schema_packages/model_system.py

```diff
@@ -492,6 +493,168 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().__init__(m_def, m_context, **kwargs)
+        # Set the name of the section
+        self.name = self.m_def.name  #! self.name here is 'ParticleCell'
+
+    def is_equal_cell(self, other) -> bool:
+        """
+        Check if the atomic cell is equal to an`other` atomic cell by comparing the `positions` and
+        the `AtomsState[*].chemical_symbol`.
+        Args:
+            other: The other atomic cell to compare with.
+        Returns:
+            bool: True if the atomic cells are equal, False otherwise.
+        """
+        if not isinstance(other, ParticleCell):
+            return False
+
+        # Compare positions using the parent sections's `__eq__` method
+        if not super().is_equal_cell(other=other):
+            return False
+
+        # Check that the `chemical_symbol` of the atoms in `cell_1` match with the ones in `cell_2`
+        check_positions = self._check_positions(
+            positions_1=self.positions, positions_2=other.positions
+        )
+        try:
+            for particle in check_positions:
+                type_1 = self.particles_state[particle[0]].particle_type
+                type_2 = other.particles_state[particle[1]].particle_type
+                if type_1 != type_2:
+                    return False
+        except Exception:
+            return False
+        return True
+
+    def get_particle_types(self, logger: 'BoundLogger') -> list[str]:
+        """
+        Get the chemical symbols of the atoms in the atomic cell. These are defined on `atoms_state[*].chemical_symbol`.
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            list: The list of chemical symbols of the atoms in the atomic cell.
+        """
+        if not self.particles_state:
+            return []
+
+        particle_labels = []
+        for particle_state in self.particles_state:
+            if not particle_state.particle_type:
+                logger.warning('Could not find `ParticlesState[*].particle_type`.')
+                return []
+            particle_labels.append(particle_state.particle_type)
+        return particle_labels
+
+    def to_particles(self, logger: 'BoundLogger') -> Optional[Particles]:
+        """
+        Generates a Particles object with the most basic information from the parsed `ParticleCell`
+        section (labels, periodic_boundary_conditions, positions, and lattice_vectors).
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Optional[Particles]): The Partilces object with the basic information from the `ParticleCell`.
+        """
+        # Initialize Partilces object with labels
+        particle_labels = self.get_particle_types(logger=logger)
+        particles = Particles(symbols=particle_labels)
+
+        # PBC
+        if self.periodic_boundary_conditions is None:
```

Comment:
I am assuming a lot of these functionalities are shared with atomic_cell. I am wondering if they can be taken up to cell, or defined consistently between the classes in some other way?

Reply:
That would make sense as far as possible. To prevent errors with CG systems, we have to check carefully for all situations where functions from `ase` are called and keep those duplicates.